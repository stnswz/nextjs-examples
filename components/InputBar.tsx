import { Box } from '@material-ui/core/';
import { makeStyles } from '@material-ui/core'
import { TextField, Button } from '@material-ui/core/';
import { useState, MouseEvent } from 'react'

interface IInputBarProps {
  sendData: (s:string) => void
}

const useStyles = makeStyles((theme) => ({
  inputBox: {
    display: 'flex',
    justifyContent: 'space-between',
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(2)

  },
  textField: {
    width: '100%'
  },
  btn: {
    marginLeft: 5,
    whiteSpace: 'nowrap'
  }
}))

export default function InputBar({sendData}: IInputBarProps) {
  const classes = useStyles()
  const [text, setText] = useState('')


  const onClick = (ev: MouseEvent) => {
    // setSearchText(text)
    sendData(text)
  }


  return (
    <Box component='div' className={classes.inputBox}>
      <TextField data-testid="text-input-field" className={classes.textField} label="Add input text" variant="outlined" size='small' value={text} onChange={ev => setText(ev.target.value)} />
      <Button className={classes.btn} onClick={onClick} variant="contained" color="primary" disableElevation>
        suchen
      </Button>
    </Box>
  )
}
