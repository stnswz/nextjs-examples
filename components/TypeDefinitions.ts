export interface IBasicCountry {
  name: string,
  code: string,
  emoji: string
}

export interface ICountry extends IBasicCountry {
  name: string,
  code: string,
  emoji: string,
  capital: string,
  currency: string,
  languages: [{
    name: string,
    native: string,
  }],
  states: [{
    name: string,
    code: string,
  }],
  continent: {
    name: string,
  }
}