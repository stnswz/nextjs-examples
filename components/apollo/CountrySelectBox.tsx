import { useState } from 'react'
import { InputLabel, MenuItem, FormControl, Select } from '@material-ui/core/'
import { makeStyles } from '@material-ui/core'
import { IBasicCountry } from '../../components/TypeDefinitions'

interface CSBProps {
  countries: Array<IBasicCountry>,
  countrySelected: (s:string) => void
}

const useStyles = makeStyles((theme) => ({
  formControl: {
    margin: theme.spacing(1),
    minWidth: 400,
    textAlign: 'left'
  }
}))

export default function CountrySelectBox({countries, countrySelected}: CSBProps) {
  const classes = useStyles()

  const [countryCode, setCountryCode] = useState('')

  const onChange = (event: any) => {
    console.log("Selected: " + event.target.value)
    setCountryCode(event.target.value);
    countrySelected(event.target.value)
  }

  return (
    <FormControl variant="outlined" className={classes.formControl}>
      <InputLabel id="country-label">Country</InputLabel>
      <Select
        labelId="country-label"
        id="country"
        value={countryCode}
        onChange={onChange}
        label="Country"
      >
        
        <MenuItem value="">
          <em>None</em>
        </MenuItem>
        
        {countries && countries.map(item => (
          <MenuItem key={item.code} value={item.code}>
            {item.emoji} {item.name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  )
}