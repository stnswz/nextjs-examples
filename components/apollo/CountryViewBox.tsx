import { Typography, Box } from '@material-ui/core/'
import { List, ListItem, ListItemIcon, ListItemText } from '@material-ui/core/'
import PublicIcon from '@material-ui/icons/Public';
import { makeStyles } from '@material-ui/core'
import { ICountry } from '../../components/TypeDefinitions'

const useStyles = makeStyles((theme) => ({
  contentBox: {
    marginTop: theme.spacing(2),
    width: '100%',
    textAlign: 'left',
    backgroundColor: '#eee',
    paddingLeft: theme.spacing(1),
  },
  countryHeadline: {
    marginLeft: -theme.spacing(1),
  },
  countryBody: {
    marginTop: theme.spacing(1),
  }
}))

export default function CountryViewBox(country: ICountry) {
  const classes = useStyles()

  function hasStates(states: Array<any>): boolean {
    return states && states.length > 0
  }

  function getLanguages(languages:Array<any>): string {
    let lang = ''
    for(let i=0; i<languages.length; i++) {
      lang += languages[i].name + ' (' + languages[i].native + ')'
      if(i < languages.length-1) {
        lang += ' / '
      }
    }
    return lang
  }

  return(
    <Box component="div" className={classes.contentBox}>
      <Typography component="h1" variant="h6" className={classes.countryHeadline}>
        {country.name} {country.emoji} 
      </Typography>
      <Typography component="h2" variant="subtitle1">
        Continent: {country.continent.name}
      </Typography>
      <Typography component="div" variant="subtitle2">
        Capital: {country.capital}
      </Typography>

      <Typography component="div" variant="body1" className={classes.countryBody}>
        Language: {getLanguages(country.languages)} <br />
        Currency: {country.currency} <br />
        States: {!hasStates(country.states) ? 'None' : <br />} 
        <List>
          {country.states.map( item => 
            <ListItem>
              <ListItemIcon>
                <PublicIcon style={{minWidth: 40}}/>
              </ListItemIcon>
              <ListItemText
                primary={item.name} 
              />
            </ListItem>
          )}
        </List>
      </Typography>
    </Box>
  )
}
