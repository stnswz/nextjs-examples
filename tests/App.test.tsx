import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer'
import InputBar from '../components/InputBar'
import CountrySelectBox from '../components/apollo/CountrySelectBox'
import CountryViewBox from '../components/apollo/CountryViewBox'
import { IBasicCountry, ICountry } from '../components/TypeDefinitions'

describe('Loading Site Components', () => {
  test('Renders InputBar', () => {
    render(<InputBar sendData={(s:string) => {}} />)
    const element = screen.getByTestId('text-input-field')
    expect(element).toBeInTheDocument()
  })
})

// Snapshot Test
describe('Apollo Site Components', () => {
  test('Renders CountrySelectBox', () => {
    const countries: Array<IBasicCountry> = [
      { name: 'Name1', code: 'Code1', emoji: 'Emoji1' },
      { name: 'Name2', code: 'Code2', emoji: 'Emoji2' }
    ]
    const componentTree = renderer.create(<CountrySelectBox countries={countries} countrySelected={(s:string) => {}} />).toJSON();
    expect(componentTree).toMatchSnapshot();
  })

  test('Renders CountryViewBox', () => {
    const country: ICountry = {
      name: 'Name1',
      code: 'Code1',
      emoji: 'Emoji1',
      capital: 'Berlin',
      currency: 'EUR',
      languages: [{
        name: 'German',
        native: 'Deutsch',
      }],
      states: [{
        name: 'StateName1',
        code: 'StateCode1',
      }],
      continent: {
        name: 'Europe',
      }
    }
    const componentTree = renderer.create(<CountryViewBox {...country} />).toJSON();
    expect(componentTree).toMatchSnapshot();
  })
})