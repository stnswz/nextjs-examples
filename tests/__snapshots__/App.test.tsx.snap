// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apollo Site Components Renders CountrySelectBox 1`] = `
<div
  className="MuiFormControl-root makeStyles-formControl-9"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
    id="country-label"
  >
    Country
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="country-label country"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      id="country"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-10 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-12"
      >
        <span>
          Country
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Apollo Site Components Renders CountryViewBox 1`] = `
<div
  className="MuiBox-root MuiBox-root-17 makeStyles-contentBox-14"
>
  <h1
    className="MuiTypography-root makeStyles-countryHeadline-15 MuiTypography-h6"
  >
    Name1
     
    Emoji1
  </h1>
  <h2
    className="MuiTypography-root MuiTypography-subtitle1"
  >
    Continent: 
    Europe
  </h2>
  <div
    className="MuiTypography-root MuiTypography-subtitle2"
  >
    Capital: 
    Berlin
  </div>
  <div
    className="MuiTypography-root makeStyles-countryBody-16 MuiTypography-body1"
  >
    Language: 
    German (Deutsch)
     
    <br />
    Currency: 
    EUR
     
    <br />
    States: 
    <br />
    <ul
      className="MuiList-root MuiList-padding"
    >
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "minWidth": 40,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            StateName1
          </span>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
